sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsClientBuilder.java

```diff
@@ -0,0 +1,170 @@
+    public MetricsClientBuilder httpClient(HttpClient httpClient) {
+        innerMetricsBuilder.httpClient(httpClient);
+        innerMetricsDefinitionsBuilder.httpClient(httpClient);
+        innerMetricsNamespaceBuilder.httpClient(httpClient);
+        return this;
+    }
+
+    /**
+     * Sets The configuration store that is used during construction of the service client.
+     * @param configuration the configuration value.
+     * @return the MetricsClientBuilder.
+     */
+    public MetricsClientBuilder configuration(Configuration configuration) {
+        innerMetricsBuilder.configuration(configuration);
+        innerMetricsDefinitionsBuilder.configuration(configuration);
+        innerMetricsNamespaceBuilder.configuration(configuration);
+        return this;
+    }
+
+    /**
+     * Sets The logging configuration for HTTP requests and responses.
+     * @param httpLogOptions the httpLogOptions value.
+     * @return the MetricsClientBuilder.
+     */
+    public MetricsClientBuilder httpLogOptions(HttpLogOptions httpLogOptions) {
+        innerMetricsBuilder.httpLogOptions(httpLogOptions);
+        innerMetricsDefinitionsBuilder.httpLogOptions(httpLogOptions);
+        innerMetricsNamespaceBuilder.httpLogOptions(httpLogOptions);
+        return this;
+    }
+
+    /**
+     * Sets The retry policy that will attempt to retry failed requests, if applicable.
+     * @param retryPolicy the retryPolicy value.
+     * @return the MetricsClientBuilder.
+     */
+    public MetricsClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        innerMetricsBuilder.retryPolicy(retryPolicy);
+        innerMetricsDefinitionsBuilder.retryPolicy(retryPolicy);
+        innerMetricsNamespaceBuilder.retryPolicy(retryPolicy);
+        return this;
+    }
+
+    /**
+     * Adds a custom Http pipeline policy.
+     * @param customPolicy The custom Http pipeline policy to add.
+     * @return the MetricsClientBuilder.
+     */
+    public MetricsClientBuilder addPolicy(HttpPipelinePolicy customPolicy) {
+        innerMetricsBuilder.addPolicy(customPolicy);
+        innerMetricsDefinitionsBuilder.addPolicy(customPolicy);
+        innerMetricsNamespaceBuilder.addPolicy(customPolicy);
+        return this;
+    }
+
+    /**
+     * Sets The TokenCredential used for authentication.
+     * @param tokenCredential the tokenCredential value.
+     * @return the MetricsClientBuilder.
+     */
+    public MetricsClientBuilder credential(TokenCredential tokenCredential) {
+        innerMetricsBuilder.credential(tokenCredential);
+        innerMetricsDefinitionsBuilder.credential(tokenCredential);
+        innerMetricsNamespaceBuilder.credential(tokenCredential);
+        return this;
+    }
+
+    /**
+     * Set the {@link ClientOptions} used for creating the client.
+     * @param clientOptions The {@link ClientOptions}.
+     * @return the {@link MetricsClientBuilder}
+     */
+    public MetricsClientBuilder clientOptions(ClientOptions clientOptions) {
```

Comment:
should these properties not be passed to the inner builder too?

Reply:
Yes, this should be passed to the inner builder but autorest is not yet updated to support `clientOptions`. So, it's a TODO item for me to fix autorest and then come back and fix this. It will be done in a separate PR.